src/storage/v3/shard_rsm.cpp

```diff
@@ -0,0 +1,443 @@
+  // with the new signature that does not require the primary label.
+  auto prim_label = acc.GetPrimaryLabel();
+
+  bool action_successful = true;
+
+  for (auto &new_vertex : req.new_vertices) {
+    if (!action_successful) {
+      break;
+    }
+    /// TODO(gvolfing) Remove this. In the new implementation each shard
+    /// should have a predetermined primary label, so there is no point in
+    /// specifying it in the accessor functions. Their signature will
+    /// change.
+    /// TODO(gvolfing) Consider other methods than converting. Change either
+    /// the way that the property map is stored in the messages, or the
+    /// signature of CreateVertexAndValidate.
+    auto converted_property_map = ConvertPropertyMap(new_vertex.properties);
+
+    // TODO(gvolfing) make sure you don't create vectors bigger than they actually should be, e.g.
+    // std::vector<memgraph::storage::v3::LabelId> converted_label_ids(new_vertex.label_ids.size());
+    // TODO(gvolfing) make sure if this conversion is actually needed.
+    std::vector<memgraph::storage::v3::LabelId> converted_label_ids;
+    converted_label_ids.reserve(new_vertex.label_ids.size());
+    for (const auto &label_id : new_vertex.label_ids) {
+      converted_label_ids.emplace_back(label_id.id);
+    }
+
+    auto result_schema = acc.CreateVertexAndValidate(
+        prim_label, converted_label_ids, ConvertPropertyVector(new_vertex.primary_key), converted_property_map);
+
+    if (result_schema.HasError()) {
+      auto &error = result_schema.GetError();
+
+      std::visit(
+          [&action_successful]<typename T>(T &&) {
+            using ErrorType = std::remove_cvref_t<T>;
+            if constexpr (std::is_same_v<ErrorType, SchemaViolation>) {
+              action_successful = false;
+              spdlog::debug("Creating vertex failed with error: SchemaViolation");
+            } else if constexpr (std::is_same_v<ErrorType, Error>) {
+              action_successful = false;
+              spdlog::debug("Creating vertex failed with error: Error");
+            } else {
+              static_assert(kAlwaysFalse<T>, "Missing type from variant visitor");
+            }
+          },
+          error);
+
+      action_successful = false;
+    }
+  }
+
+  CreateVerticesResponse resp{};
+  resp.success = action_successful;
+
+  if (action_successful) {
+    auto result = acc.Commit(req.transaction_id.logical_id);
+    if (result.HasError()) {
+      resp.success = false;
+      spdlog::debug(&"ConstraintViolation, commiting vertices was unsuccesfull with transaction id: "[req.transaction_id
+                                                                                                          .logical_id]);
+    }
+  }
+  return resp;
+}
+
+WriteResponses ShardRsm::ApplyWrite(DeleteVerticesRequest &&req) {
+  bool action_successful = true;
+  auto acc = shard_->Access();
+
+  for (auto &propval : req.primary_keys) {
+    if (!action_successful) {
+      break;
+    }
```

Comment:
Instead break you set `action_successful` to false.

Reply:
I will for the most cases, however, in the case of `ShardRsm::ApplyWrite(DeleteVerticesRequest &&req)` the intent is clearer this way, since the setting of `action_successful = false;` is happening inside a `switch` statement, where calling `break` would not effect the loop.